Sources/ClientRuntime/Networking/Http/HttpContext.swift

```diff
@@ -9,61 +9,73 @@ public class HttpContext: MiddlewareContext {
     public init(attributes: Attributes) {
         self.attributes = attributes
     }
-    // FIXME: Move all defined keys to separate file as constants to be used elsewhere
-    public func getPath() -> String {
-        return attributes.get(key: AttributeKey<String>(name: "Path"))!
-    }
 
-    public func getMethod() -> HttpMethodType {
-        return attributes.get(key: AttributeKey<HttpMethodType>(name: "Method"))!
-    }
+//    public func getAuthSchemes() -> Attributes {
+//        return attributes.get(key: AttributeKeys.authSchemes)!
+//    }
 
-    public func getEncoder() -> RequestEncoder {
-        return attributes.get(key: AttributeKey<RequestEncoder>(name: "Encoder"))!
+    public func getDecoder() -> ResponseDecoder {
+        return attributes.get(key: AttributeKeys.decoder)!
     }
 
-    public func getDecoder() -> ResponseDecoder {
-        return attributes.get(key: AttributeKey<ResponseDecoder>(name: "Decoder"))!
+    public func getEncoder() -> RequestEncoder {
+        return attributes.get(key: AttributeKeys.encoder)!
     }
 
     public func getHost() -> String? {
-        return attributes.get(key: AttributeKey<String>(name: "Host"))
+        return attributes.get(key: AttributeKeys.host)
     }
 
-    public func getServiceName() -> String {
-        return attributes.get(key: AttributeKey<String>(name: "ServiceName"))!
+    public func getHostPrefix() -> String? {
+        return attributes.get(key: AttributeKeys.hostPrefix)
     }
 
     public func getIdempotencyTokenGenerator() -> IdempotencyTokenGenerator {
-        return attributes.get(key: AttributeKey<IdempotencyTokenGenerator>(name: "IdempotencyTokenGenerator"))!
+        return attributes.get(key: AttributeKeys.idempotencyTokenGenerator)!
     }
 
-    public func getHostPrefix() -> String? {
-        return attributes.get(key: AttributeKey<String>(name: "HostPrefix"))
+    public func getIdentityResolvers() -> Attributes {
+        return attributes.get(key: AttributeKeys.identityResolvers)!
     }
 
     public func getLogger() -> LogAgent? {
-        return attributes.get(key: AttributeKey<LogAgent>(name: "Logger"))
+        return attributes.get(key: AttributeKeys.logger)
+    }
+
+    public func getMessageEncoder() -> MessageEncoder? {
+        return attributes.get(key: AttributeKeys.messageEncoder)
+    }
+
+    public func getMessageSigner() -> MessageSigner? {
+        return attributes.get(key: AttributeKeys.messageSigner)
+    }
+
+    public func getMethod() -> HttpMethodType {
+        return attributes.get(key: AttributeKeys.method)!
```

Comment:
please avoid use of forced unwrapping

Reply:
This code existed previously, it shows up as diff because I changed the order of methods in the file to be alphabetical. Avoiding use of forced unwrapping may be refactor task for later time. From what I've gathered, the code force-unwraps because by the time this method is called, this attribute HAS to have been set.

Reply:
Issue created: https://github.com/awslabs/aws-sdk-swift/issues/1168

Will pick up after finishing SRA Identity & Auth